src/components/CameraCard/CameraCard.tsx

```diff
@@ -1,76 +1,104 @@
 import { useEffect, useState } from "react";
 
 interface newPokemon {
-  id: number,
-  sprites: string,
-  name: string,
-  base_experience: number,
-  hp: number,
-  attack: number,
-  specialatt: number,
-  deffense: number
- } 
-
-const x = Math.ceil(Math.random()*150);
-console.log(x);
-
-
-const POKEMON_RANDOM_ID = `https://pokeapi.co/api/v2/pokemon/${x}`
-console.log(POKEMON_RANDOM_ID);
+  id: number;
+  sprites: string;
+  name: string;
+  base_experience: number;
+  hp: number;
+  attack: number;
+  specialatt: number;
+  deffense: number;
+}
 
 export default function CameraCard() {
+  const [pokemon, setPokemon] = useState<newPokemon | null>(null);
+  const [x, setX] = useState(Math.ceil(Math.random() * 151));
+  const [loading, setLoading] = useState(false);
 
-  const[pokemon, setPokemon] = useState <newPokemon> ()
+  function changeX() {
+    setX(Math.ceil(Math.random() * 151));
+  }
 
   useEffect(() => {
-    fetch(POKEMON_RANDOM_ID)
-      .then(res=>res.json())
-      .then(data=>{
-        const { id, sprites, name, base_experience, stats } = data
+    setLoading(true);
+    fetch(`https://pokeapi.co/api/v2/pokemon/${x}`)
+      .then((res) => res.json())
+      .then((data) => {
+        const { id, sprites, name, base_experience, stats } = data;
         const newPokemon = {
           id: id,
-          sprites: sprites.front_shiny,
+          sprites: sprites.other.dream_world.front_default,
           name: name,
           base_experience: base_experience,
           hp: stats[0].base_stat,
           attack: stats[1].base_stat,
           specialatt: stats[3].base_stat,
-          deffense: stats[2].base_stat
-        }
-        setPokemon(newPokemon)
+          deffense: stats[2].base_stat,
+        };
+        setPokemon(newPokemon);
+        setLoading(false);
       })
-  }, [])
+      .catch((error) => {
+        console.error("Error fetching Pokémon:", error);
+        setLoading(false);
+      });
+  }, [x]);
```

Comment:
Aqui te doy una sugerencia para extraer el useEffect a un customHook para que la logica se entienda mucho mejor.

```suggestion
 const usePokemonData = (pokemonId: number) => {
  const [pokemon, setPokemon] = useState<Pokemon | null>(null);
  const [error, setError] = useState<string | null>(null);

  useEffect(() => {
    const fetchPokemon = async () => {
      try {
        const res = await axios.get(`https://pokeapi.co/api/v2/pokemon/${pokemonId}`);
        setPokemon(res.data);
        setError(null);
      } catch (err) {
        console.error(err);
        setError("Failed to fetch Pokemon data. Please try again.");
      }
    };

    fetchPokemon();
  }, [pokemonId]);

  return { pokemon, error };
};

export default function CameraCard() {
  const [pokemonId, setPokemonId] = useState(1);
  const { pokemon, error } = usePokemonData(pokemonId);

  const getRandomPokemonId = () => {
    setPokemonId(Math.ceil(Math.random() * MAX_POKEMON_ID));
  };
```